model/symflower/symflower_test.go

```diff
@@ -257,3 +260,38 @@ func TestCountCharactersOfGeneratedFiles(t *testing.T) {
 		ExpectedCount: 169,
 	})
 }
+
+func TestSymflowerUnitTestsWithTimeout(t *testing.T) {
+	logOutput, logger := log.Buffer()
+	defer func() {
+		if t.Failed() {
+			t.Log(logOutput.String())
+		}
+	}()
+
+	temporaryDirectory := t.TempDir()
+	repositoryPath := filepath.Join(temporaryDirectory, "testdata", "java", "light")
+	require.NoError(t, osutil.CopyTree(filepath.Join("..", "..", "testdata", "java", "light"), repositoryPath))
+
+	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), time.Duration(1*time.Millisecond))
+	defer cancel()
+
+	start := time.Now()
+	_, err := util.CommandWithResult(ctxWithTimeout, logger, &util.Command{
+		Command: []string{
+			tools.SymflowerPath, "unit-tests",
+			"--code-disable-fetch-dependencies",
+			"--workspace", repositoryPath,
+		},
+
+		Directory: repositoryPath,
+	})
+	duration := time.Since(start)
+
+	if osutil.IsWindows() {
+		assert.ErrorContains(t, err, context.DeadlineExceeded.Error())
+	} else {
+		assert.ErrorContains(t, err, "signal: killed")
+	}
```

Comment:
Does `assert.ErrorIs` not work here?

Reply:
I guess it might. But I remember that the signal errors are actually quite nasty structurally so a string is way simpler.